### ISSUE TITLE
Support for MacOS?

### ISSUE THREAD

**Issue by wagrob** [andrie/rrd#7]
I was playing around with `Smokeping` which continously monitors my home internet speed by pings on a raspberry pi. It stores data as an RRD database I’m not familiar with. I would like to use your package to import these data into R. However, I was not able to install rrd. I get the following error:

```
* installing *source* package ‘rrd’ ...
** package ‘rrd’ successfully unpacked and MD5 sums checked
** libs
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG   -I/usr/local/include   -fPIC  -Wall -g -O2  -c registerDynamicSymbol.c -o registerDynamicSymbol.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG   -I/usr/local/include   -fPIC  -Wall -g -O2  -c rrd.c -o rrd.o
rrd.c:1:10: fatal error: 'rrd.h' file not found
#include <rrd.h>
        ^~~~~~~
1 error generated.
make: *** [rrd.o] Error 1
```

It’s not stated in the vignette if I need other stuff installed for rrd to work. Do I need to install rrdlib and other stuff? How do I get them for MacOS?

**Comment by andrie** [comment]
You can find the list of system requirements in the `SystemRequirements` field in the [package description](https://cran.r-project.org/web/packages/rrd/index.html):

```dcf
SystemRequirements:  librrd-dev, rrdtool
```

I also describe this in the [README](https://cran.r-project.org/web/packages/rrd/readme/README.html).

In short, you should install `librrd` for your platform.  I have tested this in Ubuntu as well as Red Hat, and provide the instructions for these platforms in the README.

Unfortunately I don't know whether `librrd` is available for MacOS, but hopefully this points you in the right direction.

If you discover how to do this for MacOS, please let me know so we can work on creating a better experience.

**Comment by wagrob** [comment]
Thanks. I did find librrd-dev and rrdtool for MacOS. They are easily installable using homebrew.

brew install rrdtool
brew install librrd-dev

Of note, the  latest xcode have to be installed too using
xcode-select --install
otherwise the library doesn`t compile.

**Comment by andrie** [comment]
I added instructions to the Readme.

Thank you!

**Comment by sanjaychari** [comment]
Hi,

It looks like librrd is no longer available on macOS version 14.5, and I am getting the error described in this thread while trying to run `pip3 install rrdtool`.

**Comment by 0ptimista** [comment]
> Hi,
> 
> It looks like librrd is no longer available on macOS version 14.5, and I am getting the error described in this thread while trying to run `pip3 install rrdtool`.

Same here and found another python rrdtool bindings: https://github.com/smsearcy/rrdtool-python-bindings

but still need to compile on local machine. need `rrd.h` 

I have already installed rrdtool with homebrew, but compile still fails

**Comment by 0ptimista** [comment]
> Hi,
> 
> It looks like librrd is no longer available on macOS version 14.5, and I am getting the error described in this thread while trying to run `pip3 install rrdtool`.

Hi I think I found the solution just now:
Installing with specified `rrd.h` path works!

On my machine(macOS 14.6), Once you ran `brew install rrdtool`  You can run `brew info rrdtool` to get installtion path.

Mine is :`/opt/homebrew/Cellar/rrdtool/1.8.0` and the `rrd.h` file in the `/opt/homebrew/Cellar/rrdtool/1.8.0/include`

Then in the terminal, run:
```bash
export C_INCLUDE_PATH=/opt/homebrew/Cellar/rrdtool/1.8.0/include
```
which adds `rrd.h` file to compile path

the just rerun install command, I use uv personally:
`uv pip install rrdtool-bindings` it successed.

Update 2025-02-27：
```bash
# solving rrd.h not found problem
export C_INCLUDE_PATH=/opt/homebrew/Cellar/rrdtool/1.9.0/include
# solving librrd not found problem
export LIBRARY_PATH=/opt/homebrew/Cellar/rrdtool/1.9.0/lib/
```
